blamo/lib/Unit/unit.dart

```diff
@@ -304,25 +313,23 @@ class _UnitPageState extends State<UnitPage> {
     }
     return toReturn;
   }
-  
+
   void updateUnitObject(){
-    try {
-      unitObject.depthLB = double.parse(
-          _fbKey.currentState.fields['depth-lb'].currentState.value);
-    } catch(e) {
-      unitObject.depthLB = null;
-    }
-    try {
-      unitObject.depthUB = double.parse(
-          _fbKey.currentState.fields['depth-ub'].currentState.value);
-    } catch(e) {
+    if(double.tryParse(_fbKey.currentState.fields["depth-ub"].currentState.value) != null) {
+      unitObject.depthUB = double.parse(_fbKey.currentState.fields["depth-ub"].currentState.value);
+    } else {
       unitObject.depthUB = null;
     }
-    unitObject.drillingMethods = _fbKey.currentState.fields['methods'].currentState.value;
+    if(double.tryParse(_fbKey.currentState.fields["depth-lb"].currentState.value) != null) {
+      unitObject.depthLB = double.parse(_fbKey.currentState.fields["depth-lb"].currentState.value);
+    } else {
+      unitObject.depthLB = null;
+    }
+    unitObject.drillingMethods = _fbKey.currentState.fields['methods'].currentState.value.toString();
     unitObject.tags = jsonEncode(_fbKey.currentState.fields['tags'].currentState.value);
   }
 
-  void saveObject() async{
+  Future<void> saveUnitObject() async{
```

Comment:
Changed this function name to saveUnitObject but you didn't update the function call on ln:274 thus causing a build error.